samples/net/latmon/src/main.c

```diff
@@ -0,0 +1,263 @@
+/*
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Copyright (c) 2025 Jorge Ramirez-Ortiz <jorge.ramirez@oss.qualcomm.com>
+ */
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(sample_latmon, LOG_LEVEL_DBG);
+
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/net/latmon.h>
+#include <zephyr/net/socket.h>
+
+/*
+ * Blink Control
+ * DHCP: red
+ * waiting for connection: blue
+ * sampling: green
+ */
+#define LED_WAIT_PERIOD 1000000
+#define LED_DHCP_PERIOD 500000
+#define LED_RUN_PERIOD  200000
+
+#define BLINK_THREAD_PRIORITY K_IDLE_PRIO
+#define BLINK_STACK_SIZE 4096
+static K_THREAD_STACK_DEFINE(blink_stack, BLINK_STACK_SIZE);
+
+static const struct gpio_dt_spec pulse =
+	GPIO_DT_SPEC_GET_OR(DT_PATH(zephyr_user), pulse_gpios, {0});
+static const struct gpio_dt_spec ack =
+	GPIO_DT_SPEC_GET_OR(DT_PATH(zephyr_user), ack_gpios, {0});
+
+static K_SEM_DEFINE(ack_event, 0, 1);
+
+bool dhcp_done;
```

Comment:
This shared value should at least be volatile, though optimally be exchanged for an atomic value :)